File: sea-shell-service-immutable/src/main/java/org/jesperancinha/shell/webflux/immutable/data/SeaShellAccountDto.java
package org.jesperancinha.shell.webflux.immutable.data;

import lombok.Builder;
import org.jesperancinha.shell.client.accounts.Account;

import java.math.BigDecimal;


public record SeaShellAccountDto(
        BigDecimal value,
        String currency
) {
    @Builder
    public SeaShellAccountDto {
    }

    public static SeaShellAccountDto create(Account account) {
        return SeaShellAccountDto.builder()
                .value(account.getValue())
                .currency(account.getCurrency())
                .build();
    }
}

File: sea-shell-service-immutable/src/main/kotlin/org/jesperancinha/shell/webflux/immutable/data/SeaShellAccountDto.kt
package org.jesperancinha.shell.webflux.immutable.data

import lombok.Builder
import org.jesperancinha.shell.client.accounts.Account
import java.math.BigDecimal

@JvmRecord
data class SeaShellAccountDto @Builder constructor(val value: BigDecimal, val currency: String) {
    companion object {
        fun create(account: Account?): SeaShellAccountDto {
            return SeaShellAccountDto.builder()
                .value(account!!.value)
                .currency(account.currency)
                .build()
        }
    }
}